File: 66-PlusOne/66-PlusOne.js
// Last updated: 4/29/2026, 10:35:25 AM
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
    const outputDigits = [];
    let carryout = 1;
    for(let i = digits.length - 1; i >= 0; i--) {
        const num = digits[i];
        let output = num + carryout;
        carryout = Math.floor(output / 10);
        output = output % 10;
        outputDigits.push(output)
    }
    if(carryout !== 0) {
        outputDigits.push(carryout);
    }
    return outputDigits.reverse();
};

File: 66-PlusOne/66-PlusOne.py
# Last updated: 4/29/2026, 10:35:30 AM
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        rev = digits.copy()
        print(rev)
        rev.reverse()
        print(rev)
        carryout = 0
        for i in range(len(rev)):
            sum = carryout + rev[i]
            if i == 0:
                sum += 1
            new_digit = sum % 10
            carryout = sum // 10
            print(f"{new_digit}\t{carryout}")
            rev[i] = new_digit
        if carryout == 1:
            rev.append(1)
        
        rev.reverse()
        return rev